capabilityquerier/cmd/main.go

```diff
@@ -86,14 +86,48 @@ func queryEndpoints(message []byte, args *map[string]interface{}) error {
 	return nil
 }
 
-func main() {
-	err := config.SetupConfig()
-	helpers.FailOnError("", err)
+// queryEndpointsCapabilityStatement gets an endpoint from the queue message and queries it to get the Capability Statement.
+// This function is expected to be called by the lanternmq ProcessMessages function.
+// parameter message:  the queue message that is being processed by this function, which is just an endpoint.
+// parameter args:     expected to be a map of the string "queryArgs" to the above queryArgs struct. It is formatted
+// 					   this way because queue processing is generalized.
+func queryEndpointsVersionsOperation(message []byte, args *map[string]interface{}) error {
+	// Get arguments
+	qa, ok := (*args)["queryArgs"].(queryArgs)
+	if !ok {
+		return fmt.Errorf("unable to cast queryArgs from arguments")
+	}
 
-	store, err := postgresql.NewStore(viper.GetString("dbhost"), viper.GetInt("dbport"), viper.GetString("dbuser"), viper.GetString("dbpassword"), viper.GetString("dbname"), viper.GetString("dbsslmode"))
-	helpers.FailOnError("", err)
-	log.Info("Successfully connected to DB!")
+	urlString := string(message)
+
+	jobArgs := make(map[string]interface{})
+
+	jobArgs["querierArgs"] = capabilityquerier.QuerierArgs{
+		FhirURL:      urlString,
+		Client:       qa.client,
+		MessageQueue: qa.mq,
+		ChannelID:    qa.ch,
+		QueueName:    qa.qName,
+		UserAgent:    qa.userAgent,
+		Store:        qa.store,
+	}
+
+	job := workers.Job{
+		Context:     qa.ctx,
+		Duration:    qa.jobDuration,
+		Handler:     (capabilityquerier.GetAndSendVersionsResponse),
+		HandlerArgs: &jobArgs,
+	}
 
+	err := qa.workers.Add(&job)
+	if err != nil {
+		return fmt.Errorf("error adding job to workers: %s", err.Error())
+	}
+
+	return nil
+}
+
+func setupCapQueryQueue(store *postgresql.Store, userAgent string, client *http.Client, ctx context.Context) {
```

Comment:
I feel like this and setupVersionsOperationQueue should be combined though since the only difference is the queue names and the function passed into mq.ProcessMessages, which could be passed in as function parameters